Rubberduck.VBEEditor/WindowsApi/CodePaneSubclass.cs

```diff
@@ -1,61 +1,70 @@
 ﻿using System;
-using System.Windows.Forms;
+using System.Runtime.InteropServices;
 using Rubberduck.VBEditor.Events;
 using Rubberduck.VBEditor.SafeComWrappers.Abstract;
 
 namespace Rubberduck.VBEditor.WindowsApi
 {
-    public class IntelliSenseEventArgs : EventArgs
+    //Stub for code pane replacement.  :-)
+    internal class CodePaneSubclass : FocusSource, IWindowEventProvider
     {
-        public static IntelliSenseEventArgs Shown => new IntelliSenseEventArgs(true);
-        public static IntelliSenseEventArgs Hidden => new IntelliSenseEventArgs(false);
-        internal IntelliSenseEventArgs(bool visible)
-        {
-            Visible = visible;
-        }
+        public event EventHandler<EventArgs> CaptionChanged;
+        public event EventHandler<KeyPressEventArgs> KeyDown;
+        private ICodePane _codePane;
 
-        public bool Visible { get; }
-    }
-
-    public class KeyPressEventArgs
-    {
-        public KeyPressEventArgs(IntPtr hwnd, IntPtr wParam, IntPtr lParam, char character = default)
+        public ICodePane CodePane
         {
-            Hwnd = hwnd;
-            WParam = wParam;
-            LParam = lParam;
-            Character = character;
-            if (character == default(char))
-            {
-                Key = (Keys)wParam;
-            }
-            else
+            get => _codePane;
+            set
             {
-                IsCharacter = true;
+                if (HasValidCodePane)
+                {
+                    _codePane.Dispose();          
+                }
+
+                _codePane = value;
             }
         }
 
-        public bool IsCharacter { get; }
-        public IntPtr Hwnd { get; }
-        public IntPtr WParam { get; }
-        public IntPtr LParam { get; }
-
-        public bool Handled { get; set; }
+        public bool HasValidCodePane
+        {
+            get
+            {
+                if (_codePane == null)
```

Comment:
Do we need to use `IsWrappingNullReference` here?

Reply:
Don't think so. The only thing that `CodePaneSubclass` cares about is whether *it* has a reference. In fact, it should never be used from inside the class because it would be attempting to communicate with itself from inside its own message pump.  That might need a comment\warning.